glfw/xkb_glfw.c

```diff
@@ -547,91 +548,108 @@ glfw_xkb_key_from_ime(KeyEvent *ev, bool handled_by_ime, bool failed) {
             if (sg->activeUnknownModifiers) debug("%s", format_xkb_mods(xkb, "active_unknown_mods", sg->activeUnknownModifiers));
             if (consumed_unknown_mods) { debug("%s", format_xkb_mods(xkb, "consumed_unknown_mods", consumed_unknown_mods)); }
-            else glfw_sym = clean_syms[0];
+            else xkb_sym = clean_syms[0];
             // xkb returns text even if alt and/or super are pressed
-            if ( ((GLFW_MOD_CONTROL | GLFW_MOD_ALT | GLFW_MOD_SUPER) & sg->modifiers) == 0) xkb_state_key_get_utf8(sg->state, code_for_sym, key_event.text, sizeof(key_event.text));
+            if ( ((GLFW_MOD_CONTROL | GLFW_MOD_ALT | GLFW_MOD_SUPER) & sg->modifiers) == 0) {
+              xkb_state_key_get_utf8(sg->state, code_for_sym, key_text, sizeof(key_text));
+            }
             text_type = "text";
         }
-        if ((1 <= key_event.text[0] && key_event.text[0] <= 31) || key_event.text[0] == 127) key_event.text[0] = 0;  // don't send text for ascii control codes
-        if (key_event.text[0]) { debug("%s: %s ", text_type, key_event.text); }
+        if ((1 <= key_text[0] && key_text[0] <= 31) || key_text[0] == 127) {
+          key_text[0] = 0;  // don't send text for ascii control codes
+        }
+        if (key_text[0]) { debug("%s: %s ", text_type, key_text); }
     }
-    int glfw_keycode = glfw_key_for_sym(glfw_sym);
+    int glfw_sym = glfw_key_for_sym(xkb_sym);
     bool is_fallback = false;
-    if (glfw_keycode == GLFW_KEY_UNKNOWN && !key_event.text[0]) {
+    if (glfw_sym == GLFW_KEY_UNKNOWN && !key_text[0]) {
         int num_default_syms = xkb_state_key_get_syms(sg->default_state, code_for_sym, &default_syms);
         if (num_default_syms > 0) {
-            glfw_sym = default_syms[0];
-            glfw_keycode = glfw_key_for_sym(glfw_sym);
+            xkb_sym = default_syms[0];
+            glfw_sym = glfw_key_for_sym(xkb_sym);
             is_fallback = true;
         }
     }
     debug(
         "%s%s: %d (%s) xkb_key: %d (%s)\n",
         format_mods(sg->modifiers),
-        is_fallback ? "glfw_fallback_key" : "glfw_key", glfw_keycode, _glfwGetKeyName(glfw_keycode),
-        glfw_sym, glfw_xkb_keysym_name(glfw_sym)
+        is_fallback ? "glfw_fallback_key" : "glfw_key", glfw_sym, _glfwGetKeyName(glfw_sym),
+        xkb_sym, glfw_xkb_keysym_name(xkb_sym)
     );
-    key_event.action = action; key_event.glfw_modifiers = sg->modifiers;
-    key_event.keycode = scancode; key_event.keysym = glfw_sym;
-    key_event.window_id = window->id; key_event.glfw_keycode = glfw_keycode;
-    key_event.ibus_sym = syms[0];
-    if (ibus_process_key(&key_event, &xkb->ibus)) {
-        debug("↳ to IBUS: keycode: 0x%x keysym: 0x%x (%s) %s\n", key_event.ibus_keycode, key_event.ibus_sym, glfw_xkb_keysym_name(key_event.ibus_sym), format_mods(key_event.glfw_modifiers));
+
+    glfw_ev.action = action;
+    glfw_ev.scancode = xkb_sym;
```

Comment:
Actually, this line is wrong!!! Because this is not a scancode but the native keysym instead... when it should be `xkb_keycode` instead of `xkb_sym` here (for coherence with the name of the receiving field).

This also means that `on_key_event` then kitty's keymap handling used the value of the native keysym all along.. this doesn't sound right.. do it?

Do you actually want the native keycode (scancode) or the native keysym in kitty's keyboard handling?
I would say the native keycode would make more sense, but to avoid breaking user configs, we _must_ (?) stick with keysym?

Reply:
Also note that keeping the keysym value as the "scancode" in GLFWkeyevent for linux would be inconsistent with macos, which seems to correctly put the native keycode in event struct..

Now sure what to do... break user's linux configs? (might be ok, since this feature is probably not that used.. ?)

Reply:
:warning: overkill?

We could also have a "migration" path (over N releases), where we add the keycode (good) and the keysym (bad) to the key event struct.

Then if a kitty shortcut like `ctrl+XX` matches the keysym, display a sort of toast message (in addition to doing the action) telling the user to run Kitty from a terminal, and print logs telling him to update the mapping with `ctrl+YY` instead (using the true keycode).

WDYT?

Reply:
> but which code is it

What do you mean exactly?

Reply:
Reading more, it looks like having the keyevent's "scancode" be the keysym value on linux seems to be the expected behavior:
https://github.com/kovidgoyal/kitty/blob/4ec1a8d9c3180828ae5a03cc24967b126fbec6b2/glfw/x11_window.c#L2594-L2597
So maybe I can just change the keyevent field to `native_keycode`, and document somewhere that on linux the `native_keycode` is not the `xkb_keycode` but the `xkb_keysym`..

Reply:
@Luflosi forget what I said, I think I got confused by `native_keycode`, because X11 has something called a keycode and a keysym, and it's the keysym value that is to be assigned to `native_keycode`